dkc/core/rest/folder.py

```diff
@@ -71,10 +64,12 @@ def get_access(self, folder: Folder) -> Dict[str, bool]:
         return folder.tree.get_access(self.context.get('user'))
 
     def validate(self, attrs):
+        self._validate_unique_root_name(attrs)
         self._validate_unique_file_siblings(attrs)
```

Comment:
It seems like we could just call `_get_instance_name_and_parent_id` once and pass its return values into these two private helpers, instead of calling it inside each one of them.

Reply:
I did consider this. I like the abstraction of dispatching to sub-validators that take the same `attrs` as the top-level `.validate`. `_get_instance_name_and_parent_id` is just Python, with no DB access, so it doesn't seem a big deal to call twice.

However, I don't feel strongly about this. Let me know if you think it should be changed.

Reply:
I don't have strong feelings. We can merge.